libr/anal/fcn.c

```diff
@@ -265,6 +265,63 @@ static ut64 search_reg_val(RAnal *anal, ut8 *buf, ut64 len, ut64 addr, char *reg
 
 #define gotoBeach(x) ret=x;goto beach;
 #define gotoBeachRet() goto beach;
+
+R_API void fill_args (RAnal *anal, RAnalFunction *fcn, RAnalOp *op) {
+	char* varname;
+	switch (op->stackop) {
+	case R_ANAL_STACK_NULL:
+	case R_ANAL_STACK_NOP:
+	case R_ANAL_STACK_ALIGN:
+	case R_ANAL_STACK_INC: {
+		char *esil_buf = strdup ( r_strbuf_get (&op->esil));
+		char *ptr_end = strstr (esil_buf, ",ebp,+,[4],");
+		char *addr;
+		if (!ptr_end) {
```

Comment:
dont abuse gotos, this one is not necessary, invert the conditional and nest the rest of statements